packages/dev/core/src/Physics/physicsHelper.ts

```diff
@@ -1,2 +1,1123 @@
+            return null;
+        }
+
+        const impostorObjectCenter = impostor.getObjectCenter();
+
+        return this._getHitData(mesh, impostorObjectCenter, origin);
+    }
+
+    /**
+     * Triggers affected impostors callbacks
+     * @param affectedImpostorsWithData defines the list of affected impostors (including associated data)
+     */
+    public triggerAffectedImpostorsCallback(affectedImpostorsWithData: Array<PhysicsAffectedImpostorWithData>) {
+        if (this._options.affectedImpostorsCallback) {
+            this._options.affectedImpostorsCallback(affectedImpostorsWithData);
+        }
+    }
+
+    /**
+     * Triggers affected bodies callbacks
+     * @param affectedBodiesWithData defines the list of affected bodies (including associated data)
+     */
+    public triggerAffectedBodiesCallback(affectedBodiesWithData: Array<PhysicsAffectedBodyWithData>) {
+        if (this._options.affectedBodiesCallback) {
+            this._options.affectedBodiesCallback(affectedBodiesWithData);
+        }
+    }
+    /**
+     * Disposes the sphere.
+     * @param force Specifies if the sphere should be disposed by force
+     */
+    public dispose(force: boolean = true) {
+        if (force) {
+            this._sphere.dispose();
+        } else {
+            setTimeout(() => {
+                if (!this._dataFetched) {
+                    this._sphere.dispose();
+                }
+            }, 0);
+        }
+    }
+
+    /*** Helpers ***/
+
+    private _prepareSphere(): void {
+        if (!this._sphere) {
+            this._sphere = CreateSphere("radialExplosionEventSphere", this._options.sphere, this._scene);
+            this._sphere.isVisible = false;
+        }
+    }
+
+    private _intersectsWithSphere(mesh: AbstractMesh, origin: Vector3, radius: number): boolean {
+        this._prepareSphere();
+
+        this._sphere.position = origin;
+        this._sphere.scaling = new Vector3(radius * 2, radius * 2, radius * 2);
+        this._sphere._updateBoundingInfo();
+        this._sphere.computeWorldMatrix(true);
+
+        return this._sphere.intersectsMesh(mesh, true);
+    }
+}
+
+/**
+ * Represents a gravitational field event
+ */
+class PhysicsGravitationalFieldEvent {
+    private _tickCallback: any;
+    private _sphere: Mesh;
+    private _dataFetched: boolean = false; // check if the has been fetched the data. If not, do cleanup
+
+    /**
+     * Initializes the physics gravitational field event
+     * @param _physicsHelper A physics helper
+     * @param _scene BabylonJS scene
+     * @param _origin The origin position of the gravitational field event
+     * @param _options The options for the vortex event
+     */
+    constructor(private _physicsHelper: PhysicsHelper, private _scene: Scene, private _origin: Vector3, private _options: PhysicsRadialExplosionEventOptions) {
+        this._options = { ...new PhysicsRadialExplosionEventOptions(), ...this._options };

```

Comment:
Shouldn't this be `PhysicsGravitationalFieldEventOptions`?

Reply:
I agree naming is bad. I would not change that because of back compat. same for next comment.